cpp/src/arrow/dataset/partition.cc

```diff
@@ -82,6 +82,8 @@ std::shared_ptr<Partitioning> Partitioning::Default() {
 
     std::string type_name() const override { return "default"; }
 
+    bool Equals(const Partitioning& other) const override { return false; }
```

Comment:
Are you planning to actually implement this?

Reply:
I am not sure if I misunderstood the review comment when changing the earlier code into this. I thought this must be always `false`. cc @westonpace

Reply:
I was thinking to always return false for `FunctionPartitioning`.  The default partitioning can simply return `type_name() == other.type_name()`

Reply:
Oh!. Fixing it.